### PR TITLE
Add increment and decrement statements

### PR DESCRIPTION
**Pull request by dneto0**
Fixes: #2320

### CODE REVIEW

File: wgsl/index.bs [hunk 1]
```diff
@@ -5271,6 +5271,61 @@ A phony-assignment is useful for:
...
+  <dfn for=syntax>decrement_statement</dfn> :
+
+    | [=syntax/lhs_expression=] [=syntax/minus_minus=]
+</div>
+
+The expression must evaluate to a reference with an [=integer scalar=] [=store type=] and [=access/read_write=] [=access mode=].
+
+<table class='data'>
+  <thead>
+    <tr><th>Precondition<th>Statement<th>Description
```
Comment (dneto0) [inline wgsl/index.bs:5295]: This is a lot of words.

Maybe take the approach in #2345, depending on feedback there.

**Comment by dneto0** [comment]
Made this based on #2345, and defined both in terms of += and -=

File: wgsl/index.bs [hunk 2]
```diff
@@ -8240,8 +8435,58 @@ A <dfn>syntactic token</dfn> is a sequence of special characters, used:
...
+    | `'>>='`
+</div>
+<div class='syntax' noexport='true'>
+  <dfn for=syntax>shift_left_equal</dfn> :
+
+    | `'<<='`
+</div>
 
-Note: The `MINUS_MINUS` and `PLUS_PLUS` tokens are reserved, i.e. they are not used in any grammar productions.
+Note: The [=syntax/plus_plus=] and [=syntax/minus_minus=] tokens are reserved, i.e. they are not used in any grammar productions.
```
Comment (alan-baker) [inline wgsl/index.bs:8489]: Forgot to remove this?

**Comment by kvark** [comment]
Needs a rebase

**Review by alan-baker — APPROVED** [review]

**Review by kvark — APPROVED** [review]

**Pull request merged**